### PR TITLE
Add Max Clamping

### PR DESCRIPTION
**Pull request by ConnorVG**
In situations where you need to add custom max clamping (such as `< 2^16` etc).

### CODE REVIEW

File: src/Optimus.php [hunk 1]
```diff
@@ -115,4 +122,14 @@ public function setMode($mode)
 
         static::$mode = $mode;
     }
+
+    /**
+     * Set the maximum integer.
+     *
+     * @param int $max
+     */
+    public function setMax($max)
```
Comment (jenssegers) [inline src/Optimus.php]: Not sure if I would add this method. Does it make sense to change the maximum after an instance has been created?

File: src/Energon.php [hunk 2]
```diff
@@ -44,26 +44,48 @@ public static function generate($prime = null)
     }
 
     /**
+     * Generates a set of numbers ready for use (with a custom max cap).
+     *
+     * @param int $max
+     *
+     * @return array
+     */
+    public static function generateMax($max)
```
Comment (jenssegers) [inline src/Energon.php]: Can we remove this method and add a parameter to the existing `generate` method?

Comment (ConnorVG) [inline src/Energon.php]: We could but then we'd always, when using generate and not using a predefined prime, have to pass `null, $max`. Are you sure?

File: src/Optimus.php (hunk 1 above)
Comment (ConnorVG) [inline src/Optimus.php]: Probably not, I took this from a suggestion in an issue I found.

File: src/Energon.php (hunk 2 above)
Comment (jenssegers) [inline src/Energon.php]: Not sure if a lot of people are using `generate` with a predefined prime. Maybe we can release a version without that option and replace it with `$max`.

Comment (ConnorVG) [inline src/Energon.php]: I agree completely, want me to drop it?

**Pull request closed without merging**